cilium-cli/cli/version.go

```diff
@@ -16,9 +17,13 @@ import (
 )
 
 func getLatestStableVersion() string {
-	resp, err := http.Get("https://raw.githubusercontent.com/cilium/cilium/main/stable.txt")
+	client := &http.Client{
+		Timeout: 10 * time.Second,
```

Comment:
Not sure I am keen on canning the timeout. 
Perhaps best to have a cli arg for it and default to 10s?

Reply:
If we want to add a timeout parameter to the version subcommand, it should apply to the RunE function, similar to the following logic:
```go
	ctx, cancel := context.WithTimeout(context.Background(), timeout*time.Second)
	defer cancel()

	done := make(chan int, 1)

	go func() {
		fmt.Printf("cilium-cli: %s compiled with %v on %v/%v\n", defaults.CLIVersion, runtime.Version(), runtime.GOOS, runtime.GOARCH)
		fmt.Printf("cilium image (default): %s\n", defaults.Version)
		fmt.Printf("cilium image (stable): %s\n", getLatestStableVersion())

		if clientOnly {
			done <- 1
			return
		}

		version, err := k8sClient.GetRunningCiliumVersion(helmReleaseName)
		if err != nil {
			fmt.Printf("cilium image (running): unknown. Unable to obtain cilium version. Reason: %s\n", err.Error())
		} else {
			fmt.Printf("cilium image (running): %s\n", version)
		}
		done <- 1
	}()
	
	select {
	case <-done:
	case <-ctx.Done():
		fmt.Printf("Operation timed out after %d seconds", timeout)
	}
	return nil
```
Are we sure we want to do it this way? I feel it's unnecessary.